src/arkode/arkode_erkstep.c

```diff
@@ -1144,6 +1208,340 @@ int erkStep_GetOrder(ARKodeMem ark_mem)
   return step_mem->q;
 }
 
+/*---------------------------------------------------------------
+  Utility routines for ERKStep to serve as an MRIStepInnerStepper
+  ---------------------------------------------------------------*/
+
+/*------------------------------------------------------------------------------
+  ERKStepCreateMRIStepInnerStepper
+
+  Wraps an ERKStep memory structure as an MRIStep inner stepper.
+  ----------------------------------------------------------------------------*/
+
+int ERKStepCreateMRIStepInnerStepper(void* inner_arkode_mem,
```

Comment:
The next couple hundred lines of code are essentially duplicating what's in ARKstep. Most of the inner stepper functions implemented below are calling generic `ARKode*` functions rather than `ERKStep*` functions. I think this is a sign we should make a single `ARKodeCreateMRIStepInnerStepper` function that works across ARKODE. I think the only functionality that's not fully supported across the steppers is forcing (and maybe some of the new error accumulation stuff). These can be optional function pointers provided to the ARKODE infrastructure if supported (ARKstep and ERKstep).

Reply:
Given that the setforcing, applyforcing, and fullrhs functions differ between each stepper, I think that the simplest approach would be to create ARKODE-level utility routines for the others (e.g., reset, get/reset accumulated error, setrtol).  Then we could still have stepper-specific "CreateMRIStepInnerStepper" utility routines, but those would attach the ARKODE-level utility routines were appropriate.